eldritch/events.py

```diff
@@ -1037,6 +1038,57 @@ def GainAllyOrReward(character, ally: str, reward: Event):
   return PassFail(character, has_ally, gain_ally, reward)
 
 
+class SellChosen(Event):
+
+  def __init__(self, character, choice, discount_type="fixed", discount=0):
+    assert discount_type in {"fixed", "rate"}
+    self.character = character
+    self.choice = choice
+    self.sellable = None
+    self.discount_type = discount_type
+    self.discount = discount
+    self.prices = []
+    self.resolved = False
+    self.sold = None
+
+  def resolve(self, state):
+    if self.resolved:
+      # This should never happen??
+      return True
+    for card in self.choice.chosen:
+      price = self.discounted_price(card)
+      self.prices.append(price)
```

Comment:
Small nitpick - I recommend
```
self.prices = [self.discounted_price(card) for card in self.choice.chosen]
```
before the for loop. Or alternatively, calling self.prices.clear() before entering the loop. If at some point in the future this event were modified so that resolve might be called multiple times, it would be good to have the calculation of self.prices not re-extend the list.